tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -781,6 +781,295 @@ int main(int argc, char **argv)
+                        &s2n_dhe_rsa_with_chacha20_poly1305_sha256,     /* 0xCC,0xAA */
+                    };
+
+                    const struct s2n_cipher_preferences cipher_preference_with_tls12_chacha20_and_boosting = {
+                        .count = s2n_array_len(cipher_suites_with_tls12_chacha20),
+                        .suites = cipher_suites_with_tls12_chacha20,
+                        .allow_chacha20_boosting = true
+                    };
+
+                    const struct s2n_security_policy test_tls12_chacha20_and_boosted = {
+                        .minimum_protocol_version = S2N_SSLv2,
+                        .cipher_preferences = &cipher_preference_with_tls12_chacha20_and_boosting,
+                        .kem_preferences = &kem_preferences_null,
+                        .signature_preferences = &s2n_signature_preferences_20201021,
+                        .ecc_preferences = &s2n_ecc_preferences_test_all,
+                    };
+
+                    conn->security_policy_override = &test_tls12_chacha20_and_boosted;
+                    conn->kex_params.server_ecc_evp_params.negotiated_curve = s2n_all_supported_curves_list[0];
+                    conn->client_protocol_version = S2N_TLS12;
+                    conn->actual_protocol_version = S2N_TLS12;
+                    conn->server_protocol_version = S2N_TLS12;
+
+                    uint8_t ecdhe_chacha20_boosted_wire[] = {
+                        TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+                        TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+                    };
+
+                    uint8_t count = sizeof(ecdhe_chacha20_boosted_wire) / S2N_TLS_CIPHER_SUITE_LEN;
+
+                    EXPECT_SUCCESS(s2n_set_cipher_as_tls_server(conn, ecdhe_chacha20_boosted_wire, count));
+                    /* Server and client have chacha20 boosting enabled. Server negotiates its most preferred
+                     * chacha20 cipher suite (which happens to also be the client's most preferred)
+                     */
+                    EXPECT_EQUAL(conn->secure->cipher_suite, &s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256);
+
+                    uint8_t ecdhe_rsa_chacha20_boosted_wire[] = {
+                        TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                        TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+                        TLS_CHACHA20_POLY1305_SHA256,
+                        TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                    };
+                    count = sizeof(ecdhe_rsa_chacha20_boosted_wire) / S2N_TLS_CIPHER_SUITE_LEN;
+                    EXPECT_SUCCESS(s2n_set_cipher_as_tls_server(conn, ecdhe_rsa_chacha20_boosted_wire, count));
+                    /* Server and client have chacha20 boosting enabled. Server negotiates its most preferred
+                     * chacha20 cipher suite (which happens to NOT be the client's most preferred)
+                     */
```

Comment:
Please separate out these tests so that it's clear where one ends and the next begins. We generally use {} for that, witch each test case labeled with a comment. Like:
```
/* with chacha20 boosting: server and client have same most-preferred chacah20 cipher suite */
{

}

/* with chacha20 boosting: server and client have different most-preferred chacah20 cipher suite */
{

}

/* with chacha20 boosting: server chooses lower priority chacha20 cipher suite because most preferred is invalid for connection */
{

}
```